src/gwt/src/org/rstudio/studio/client/workbench/views/files/FilesUpload.java

```diff
@@ -67,22 +67,27 @@ public void execute()
          new OperationWithInput<PendingFileUpload>() {
             public void execute(PendingFileUpload pendingUpload)
             {
-               // confirm overwrites if necessary
+               Boolean unzipFound = pendingUpload.getUnzipFound();
+               Boolean isZip = pendingUpload.getIsZip();
                FileUploadToken token = pendingUpload.getToken();
-               if (pendingUpload.getOverwrites().length() > 0)
+
+               // confirm unzip is installed
+               if (unzipFound == Boolean.FALSE && isZip == Boolean.TRUE)
                {
+                  // Warn user unzip is not installed
                   globalDisplay_.showYesNoMessage(
-                       MessageDialog.WARNING, 
-                       "Confirm Overwrite",
-                       confirmFileUploadOverwriteMessage(pendingUpload), 
-                       false, 
-                       completeFileUploadOperation(token, true), 
-                       completeFileUploadOperation(token, false), 
-                       false);
+                          MessageDialog.WARNING,
+                          "Unzip not found on PATH",
+                          "Unzip was not found on users PATH. Would you like to upload the zip archive without unzipping?",
+                          false,
+                          checkForFileUploadOverwrite(pendingUpload, token),
+                          () -> {},
+                          true
+                          );
                }
-               else
-               {
-                  completeFileUploadOperation(token, true).execute();
+               else {
+                  // Upload and warn of overwrites, if any
+                  checkForFileUploadOverwrite(pendingUpload, token).execute();
                }
```

Comment:
Is there any cleanup I need to do here at the end, particularly if the user cancels before `checkForFileUploadOverwrite()` is called?

Reply:
I think the main requirement here is that `completeFileUploadOperation(token, false)` is called if the upload is cancelled.

Reply:
Fixed. All control paths will eventually call `completeFileUploadOperation()`